airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/IntegrationCommand.kt

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import io.airbyte.cdk.core.operation.Operation
+import io.airbyte.cdk.core.operation.OperationType
+import io.airbyte.cdk.integrations.base.AirbyteTraceMessageUtility
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.integrations.util.ApmTraceUtils
+import io.airbyte.cdk.integrations.util.ConnectorExceptionUtil
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Value
+import jakarta.inject.Inject
+import jakarta.inject.Named
+import java.nio.file.Path
+import java.util.Optional
+import java.util.function.Consumer
+import picocli.CommandLine
+
+private val logger = KotlinLogging.logger {}
+
+/** CLI implementation that invokes the requested operation for the connector. */
+@CommandLine.Command(
+    name = "airbyte-connector",
+    description = ["Executes an Airbyte connector"],
+    mixinStandardHelpOptions = true,
+    header =
+        [
+            "@|magenta     ___    _      __          __       |@",
+            "@|magenta    /   |  (_)____/ /_  __  __/ /____   |@",
+            "@|magenta   / /| | / / ___/ __ \\/ / / / __/ _   |@",
+            "@|magenta  / ___ |/ / /  / /_/ / /_/ / /_/  __/  |@",
+            "@|magenta /_/  |_/_/_/  /_.___/\\__, /\\__/\\___/|@",
+            "@|magenta                    /____/              |@",
+        ],
+)
+@SuppressFBWarnings(
+    value = ["NP_NONNULL_RETURN_VIOLATION", "NP_OPTIONAL_RETURN_NULL"],
+    justification = "Uses dependency injection",
+)
+class IntegrationCommand : Runnable {
+    @Value("\${micronaut.application.name}") lateinit var connectorName: String
+
+    @Inject lateinit var operations: List<Operation>
+
+    @Inject
+    @Named("outputRecordCollector")
+    lateinit var outputRecordCollector: Consumer<AirbyteMessage>
+
+    @CommandLine.Parameters(
+        index = "0",
+        description =
+            [
+                "The command to execute (check|discover|read|spec|write)",
+                "\t check - checks the config can be used to connect",
+                "\t discover - outputs a catalog describing the source's catalog",
+                "\t read - reads the source and outputs messages to STDOUT",
+                "\t spec - outputs the json configuration specification",
+                "\t write - writes messages from STDIN to the integration",
```

Comment:
s/integration/destination?

Consider splitting this class into SourceCommand and DestinationCommand sealed classes?

Reply:
@postamar I am trying to avoid having the harness know about sources or destinations, which is what the current `IntegrationRunner` does.  That has lead to some weird selection logic in there, when it really is the responsibility of the executors to know which they support.  The design idea here is that the actual operation executors would encapsulate that difference and/or use conditional loading logic to be available for one type of connector or the other.

Reply:
OK, makes sense. Let's run with this and see where it leads us.